File: readability/readability.py
# Getting the text from the user
text = input("Text: ")
# Defining variables and setting values
letters = sentences = 0.0
words = text.count(" ") + 1.0
sentences = float(text.count(".") + text.count("!") + text.count("?"))
for char in text:
    if char.isalpha():
        letters += 1.0
# Calculating the index
l = letters * 100 / words
s = sentences * 100 / words
index = round(l * 0.0588 - s * 0.296 - 15.8)
# Printing the result:
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Getting the text from the User
    string text = get_string("Text: ");
    // Defining the variables
    float letters = 0, words = 1, sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char digit = toupper(text[i]);
        // Checking letters, words and sentences
        if ('A' <= digit && 'Z' >= digit)
        {
            letters++;
        }
        else if (digit == ' ')
        {
            words++;
        }
        else if (digit == '.' || digit == '?' || digit == '!')
        {
            sentences++;
        }
    }
    // Calculating the index
    long double l = letters * 100 / words, s = sentences * 100 / words;
    int index = round(l * 0.0588 - s * 0.296 - 15.8);
    // Output
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}